src/client.js

```diff
@@ -0,0 +1,442 @@
+import { format, parse } from 'url';
+import http from 'http';
+import https from 'https';
+import superagent from 'superagent';
```

Comment:
Turns out `got` has some issues and doesn't work in the browser, and really I just needed to ditch superagent-promise, not superagent.